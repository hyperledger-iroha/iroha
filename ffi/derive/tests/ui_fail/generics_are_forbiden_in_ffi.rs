use getset::Getters;
use iroha_ffi::{ffi_export, FfiType};

#[ffi_export]
fn freestanding<T>(v: T) -> T {
    v
}

#[derive(Getters, FfiType)]
#[getset(get = "pub")]
#[ffi_export]
pub struct FfiStruct<T> {
    inner: T,
}

fn main() {}
