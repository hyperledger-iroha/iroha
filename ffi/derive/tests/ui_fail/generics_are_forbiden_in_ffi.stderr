error: Generics are not supported
 --> tests/ui_fail/generics_are_forbiden_in_ffi.rs:5:21
  |
5 | pub fn freestanding<T>(v: T) -> T {
  |                     ^

error: Generics on derived methods not supported
  --> tests/ui_fail/generics_are_forbiden_in_ffi.rs:12:21
   |
12 | pub struct FfiStruct<T> {
   |                     ^^^
